PrestaShop/PrestaShop
# Simplify rendering of forms for 1.7.7 pages

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplify rendering of forms for 1.7.7 pages
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/16891
| How to test?  | See below

## How to test

The best way to test is
- open the modified page with the branch `1.7.7.x`
- open the same form page with this pull request branch
=> **Their display and HTML/JS behavior should be the same.** We modified the code but not the behavior.
Also if possible, run browser based UI tests.

Nice to have: check if other not related forms to this  correctly - regression UI automated tests would be perfect. This would be nice because we changed the general templates elements rendering system (for example: how country state is rendered - appear/disappear).

Also some bug possibility might have been introduced because we started simplifying with form_rest and then refactored to use form_widget for rendering instead.

## List of modified pages (all 1.7.7.x pages)

- Catalog > Files add/edit form (migrated by @RaimondasSapola) (see form)
- Catalog > Supplier add/edit form (migrated by @zuk3975) (see form)
- "Improve > International > Taxes" options form
- Form type for tax add/edit
- Customers > Addresses -> add/edit form
- Customer Service > Order Messages > Add new / edit order message form
- Credit Slips (can only be added, no edited) form


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/16938)
